### ISSUE TITLE
GNOME version change

### ISSUE THREAD

**Issue by mattst88** [repology/repology-rules#608]
**Projects affected**

Most GNOME packages—anything that changed from a `3.38.2` to a `40.2` version scheme.

[nautilus](https://repology.org/project/nautilus/versions) is a good example.

**Observed behavior**

42.1.1 is detected as a "devel version" (and as a result Gentoo's webapp that queries repology misidentified the latest version, apparently because it ignores "devel versions").

The cause is the rule in `900.version-fixes/n.yaml`:

`- { name: nautilus,                    verpat: "[0-9]+\\.[0-9]*[13579]\\..*",              devel: true }`

The regex is recognizing X.Y.Z where Y is odd as a devel version, because in the past GNOME versioning worked that way. E.g. 3.37.2 was a devel version because 37 is odd, but 3.38.2 was a stable version because 38 is even.

But now it misidentifies 42.1.1 as a devel version because the middle component is odd.

**Expected behavior**

The odd devel-versioning is gone in GNOME 40+, replaced by `.alpha`, `.beta`, and `.rc` versions.

- 42.1.1 should be detected as a stable version.
- 42.beta should be detected as a devel version

I'm happy to submit a PR removing the current (outdated) rules, if you think that's appropriate. I don't suspect there's much value in retaining the ability to recognize outdated devel versions, but if that's somehow easy to do, advise me how and I'll give it a try.

**Comment by mattst88** [comment]
It was suggested to me that we change the regex like so:

````diff
-- { name: nautilus, verpat: "[0-9]+\\.[0-9]*[13579]\\..*", devel: true }
+- { name: nautilus, verpat: "[1-3]\\.[0-9]*[13579]\\..*", devel: true }
````

Or just remove the `+`:

````diff
-- { name: nautilus, verpat: "[0-9]+\\.[0-9]*[13579]\\..*", devel: true }
+- { name: nautilus, verpat: "[0-9]\\.[0-9]*[13579]\\..*", devel: true }
````

I'll look through GNOME versions and see if I see anything that these new regular expressions would not handle properly.

**Comment by AMDmi3** [comment]
These rules may be removed completely, as devel statuses for non-latest versions are not distinguished, and for new versions devel status will be set based on alpha/beta/rc suffix which is done by default. I'm removing the rule for nautilus, please list other affected packages or submit a PR.

**Comment by mattst88** [comment]
Thank you. I will open a PR (And thank you for repology and all your work on it. It's fantastic!)
